### DISCUSSION TITLE
Dynamic bandwidth support

### DISCUSSION THREAD

**Discussion by gehee** [svpcom/wfb-ng#311] [Ideas]
I am new to this so please tell me if I understand incorrectly but would it be possible to implement dynamic bandwidth as RSSI get better or worse by varying mcs_index ?
Can the rx tell the tx to update it's mcs_index using a feedback loop? would it be fast enough to work?

**Comment by svpcom** [comment]
It is possible in theory, but it requires bidirectional link and able to control camera codec bitrate, because if you want to increase/decrease bandwidth you will need to adjust camera bitrate too

**Reply by gehee** [reply]
Yeah that's the end goal for this; isn't there a bidirectional communication happening already ? I guess it's not at the rf link level?
What would need to change in wfb-ng for this to work?

**Reply by svpcom** [reply]
Bidirectional communication supported, but not required. For example  video stream is unidirectional transfer by default.
Need to add code to control plane for link speed negotiation and add api for dynamic modulation change to data plane
